home/models.py

```diff
@@ -41,24 +62,49 @@ def render_basic(self, value, context=None):
         pass
 
 
-def get_choices():
+def get_valid_profile_values(field):
     if SiteSettings.objects.first():
-        return [
-            (choice, choice)
-            for choice in SiteSettings.objects.first().content_variations_options
-        ]
+        profile_values = {}
+        for profile_block in SiteSettings.objects.first().profile_field_options:
```

Comment:
we shouldn't get first site settings objects but correct settings for site. I think you can do SiteSettings.for_site(site_object) iirc or you can get site for request